src/transformers/trainer.py

```diff
@@ -2859,7 +2865,13 @@ def _save_checkpoint(self, model, trial, metrics=None):
             metric_to_check = self.args.metric_for_best_model
             if not metric_to_check.startswith("eval_"):
                 metric_to_check = f"eval_{metric_to_check}"
-            metric_value = metrics[metric_to_check]
+            try:
+                metric_value = metrics[metric_to_check]
+            except KeyError as exc:
+                raise KeyError(
+                    f"The `metric_for_best_model` training argument is set to '{metric_to_check}', which is not found in the evaluation metrics. "
```

Comment:
I'm not extremely sold on this phrasing, because `metric_for_best_model` is not necessarily set to `metric_to_check`, because the latter can be e.g. `eval_blabla` when the former was `blabla`.

Open to suggestions here!